### PR TITLE
assert that string starts with a given string

### PR DESCRIPTION
**Pull request by fredr**
Added startWith() for asserting that a string starts with a given string.

### CODE REVIEW

**Comment by travisjeffery** [comment]
what's the use-case? when are you doing this often enough that it makes sense to not use `'"foobar".should.match(/^foo/)`

**Comment by btd** [comment]
Agreed @travisjeffery , i think we can close this.

**Comment by fredr** [comment]
Well, should.js is an expressive library for asserting, and imo `'http://www.example.com/something/something'.should.startWith('http://www.example.com')` is more readable than doing a regexp. 

But yes, it is possible to regexp matches that does the same thing. This PR is just a suggestion

**Comment by gabrielf** [comment]
I agree that startsWith('...') is more readable than using a regexp in many cases. Especially when looking for characters that must be escaped in a regexp such as:

someUrl.should.startWith('http://') compared to someUrl.should.match(/^http:\/\//)

**Comment by btd** [comment]
> someUrl.should.startWith('http://') compared to someUrl.should.match(/^http:\/\//)

this definitely makes sense

File: test/should.test.js [hunk 1]
```diff
@@ -419,6 +419,27 @@ module.exports = {
     ({length: 10}).should.have.ownProperty('length').equal(10);
   },
 
+  'test startsWith()': function() {
```
Comment (btd) [inline test/should.test.js]: Could you set it to `startWith` as everywhere else.

File: lib/should.js [hunk 2]
```diff
@@ -489,6 +489,19 @@ Assertion.prototype = {
...
+   * @param {String} str
+   * @param {String} desc
+   * @api public
+   */
+
+  startWith: function(str, desc) {
+    this.assert(0 === this.obj.indexOf(str)
+    , function() { return 'expected ' + this.inspect + ' to start with ' + i(str) + (desc ? " | " + desc : "") }
+    , function() { return 'expected ' + this.inspect + ' to not start with ' + i(str) + (desc ? " | " + desc : "") });
+  },
```
Comment (btd) [inline lib/should.js:507]: `return this` for chaining.

**Comment by travisjeffery** [comment]
@gabrielf yeah that's a good use-case/example

File: test/should.test.js (hunk 1 above)
Comment (fredr) [inline test/should.test.js]: nice catch!

**Comment by fredr** [comment]
@btd hmm, did I do something strange while merging here? or does the PR look ok?

**Comment by btd** [comment]
@travisjeffery looks ok for me, what do you think?

**Comment by travisjeffery** [comment]
Good, just wanted to squash those commits.

**Comment by travisjeffery** [comment]
Ayee, that boinked attribution. One sec.

**Comment by fredr** [comment]
:beers:

**Comment by gabrielf** [comment]
Nice!

**Pull request closed without merging**